packages/editor/src/store/actions.js

```diff
@@ -195,23 +195,38 @@ export const savePost =
 				),
 			content,
 		};
-		dispatch( { type: 'REQUEST_POST_UPDATE_START', options } );
-		await registry
-			.dispatch( coreStore )
-			.saveEntityRecord(
-				'postType',
-				previousRecord.type,
+		let error;
+		try {
+			edits = await applyFilters(
+				'editor.__unstablePreSavePost',
 				edits,
 				options
 			);
+		} catch ( err ) {
+			error = err;
+		}
 
-		let error = registry
-			.select( coreStore )
-			.getLastEntitySaveError(
-				'postType',
-				previousRecord.type,
-				previousRecord.id
-			);
+		dispatch( { type: 'REQUEST_POST_UPDATE_START', options } );
```

Comment:
Let's move this "post update start" call before the `await` call for the pre-save filter. When the filter is being applied, some async processing is already happening and we want the `isSavingPost` selector to return `true`. One of the visible effects is that the "Save" button is disabled immediately after clicking, not after a delay.

Reply:
moved in 4ead1ff2876457b9007f26e7d200e6cb9c7a0760